lib/router/case.js

```diff
@@ -1,25 +1,15 @@
 const { Router } = require('express');
-
 const Case = require('../models/case');
 
-module.exports = ({ hooks }) => {
+module.exports = ({ hooks, caseDecorator }) => {
 
   const router = Router();
 
-  router.param('case', (req, res, next, id) => {
-    Case.find(id)
-      .then(model => {
-        req.case = model;
-        next();
-      })
-      .catch(next);
-  });
-
-  router.get('/:case', (req, res, next) => {
+  router.get('/', caseDecorator, (req, res, next) => {
     res.respondWith(req.case);
   });
 
-  router.put('/:case', (req, res, next) => {
+  router.put('/', (req, res, next) => {
```

Comment:
We really should be applying the decorator to the results being sent back by `PUT` calls too.